fendermint/actors/chainmetadata/src/actor.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use cid::Cid;
+use fil_actors_runtime::actor_dispatch;
+use fil_actors_runtime::actor_error;
+use fil_actors_runtime::builtin::singletons::SYSTEM_ACTOR_ADDR;
+use fil_actors_runtime::runtime::{ActorCode, Runtime};
+use fil_actors_runtime::ActorError;
+use std::collections::VecDeque;
+
+use crate::{ConstructorParams, Method, State};
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(Actor);
+
+pub struct Actor;
+
+impl Actor {
+    fn constructor(rt: &impl Runtime, params: ConstructorParams) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_is(std::iter::once(&SYSTEM_ACTOR_ADDR))?;
+
+        let state = State {
+            blockhashes: VecDeque::new(),
```

Comment:
Have you considered storing this in an AMT? It would allow structural sharing, whereas this will always copy the whole list of hashes.

Reply:
I think it supports popping values via [delete](https://github.com/filecoin-project/ref-fvm/blob/8a93ecdf43892cc15ba997d9e2c742b272a48fd7/ipld/amt/src/amt.rs#L329). A short example of using this type is in the docstring in https://github.com/consensus-shipyard/ipc/blob/main/ipc/types/src/amt.rs